core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -190,14 +190,19 @@ private Http2ConnectionHandler newHttp2ConnectionHandler(ChannelPipeline pipelin
 
     private Http2Settings http2Settings() {
         final Http2Settings settings = new Http2Settings();
-        final int initialWindowSize = config.http2InitialWindowSize();
+        final int initialWindowSize = config.http2InitialStreamWindowSize();
         if (initialWindowSize != Http2CodecUtil.DEFAULT_WINDOW_SIZE) {
             settings.initialWindowSize(initialWindowSize);
         }
-        final int maxStreamsPerConnection = config.http2MaxStreamsPerConnection();
-        if (maxStreamsPerConnection != Integer.MAX_VALUE) {
-            settings.maxConcurrentStreams(maxStreamsPerConnection);
+
+        final int maxFrameSize = config.http2MaxFrameSize();
+        if (maxFrameSize != Http2CodecUtil.DEFAULT_MAX_FRAME_SIZE) {
+            settings.maxFrameSize(maxFrameSize);
         }
+
+        // Not using the value greater than 2^31-1 because some HTTP/2 client implementations use a signed
+        // 32-bit integer to represent an HTTP/2 SETTINGS parameter value.
+        settings.maxConcurrentStreams(Math.min(config.http2MaxStreamsPerConnection(), Integer.MAX_VALUE));
```

Comment:
I thought I left a comment here but can't find it anymore - sorry if I double-comment.

If we clamp to int anyways shouldn't we leave the type as int instead of long?